android/expoview/src/main/java/host/exp/exponent/RNObject.java

```diff
@@ -46,9 +47,11 @@ public boolean isNotNull() {
     return mInstance != null;
   }
 
+  @SuppressWarnings("ConstantConditions")   // required for "unversioned" flavor check
   public RNObject loadVersion(String version) {
     try {
-      if (version.equals(UNVERSIONED)) {
+
+      if (version.equals(UNVERSIONED) || BuildConfig.FLAVOR.equals("unversioned")) {
```

Comment:
I made this change to directly reflect the iOS behavior: https://github.com/expo/expo/pull/11999/files#diff-a6fe7dbdc91453a98e00e665c550fb614e5fc0d380de6fc1f08db8d94b8c4128 introduced by @tsapeta in #11999

Maybe a warning would be sufficient (`sdkVersion != unversioned && flavor == unversioned`), but if we just display it in logcat, it would quickly get lost in hundreds of other messages there (unless it's displayed with each versioned call).

Reply:
@esamelson I think it's not so common case when we need to test versioned code in our day-to-day development (except versioned QA). When you're doing some work on unversionable code used across all versioned SDKs, then I think it's best for you to just build versioned flavor and test it there. I believe having different icon, app name and bundle ID are clear enough for everyone.

Also, it's worth to note that all those things have one more intention — to make it easier to contribute to Expo. For people from outside the team, there is almost no reason to build the client with versioned code. 
Using unversioned code for all SDKs has a lot of advantages and one of them is that you can easily open each snack with the "currently-in-development" code and test if old projects (or demos provided in the issues) are working fine with the changes you made. We wouldn't need to explain "if you want to test your changes, open the snack, download it, modify sdkVersion in app.json and start the app" — why do we complicate it so much?

To be honest, after splitting it up on iOS, I started testing things with the already published NCL app (the most recent one for non-unversioned SDK) without even running `expo start` in NCL. Can it be easier for simple changes? This finally makes the development pleasant.